babel/messages/catalog.py

```diff
@@ -286,9 +289,9 @@ def python_format(self) -> bool:
 
         :type:  `bool`"""
         ids = self.id
-        if not isinstance(ids, (list, tuple)):
-            ids = [ids]
-        return any(PYTHON_FORMAT.search(id) for id in ids)
+        if isinstance(ids, (list, tuple)):
+            return any(PYTHON_FORMAT.search(id) for id in ids)
```

Comment:
```suggestion
            return any(filter(PYTHON_FORMAT.search, ids))
```

Though also the list comprehension inlining from [PEP 709](https://peps.python.org/pep-0709/) may be faster:

```suggestion
            return any([PYTHON_FORMAT.search(id) for id in ids])
```